File: wheystore/wheystore.Web/Modules/SanPhamDB/SanPhamHuongVi/SanPhamHuongViColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.SanPhamHuongVi")]
[BasedOnRow(typeof(SanPhamHuongViRow), CheckNames = true)]
public class SanPhamHuongViColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int? Id { get; set; }
    public string TenHuongVi { get; set; }
    public bool IsDefault { get; set; }
}

File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/SanPhamHuongViColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SanPhamHuongViRow } from "./SanPhamHuongViRow";

export interface SanPhamHuongViColumns {
    Id: Column<SanPhamHuongViRow>;
    TenHuongVi: Column<SanPhamHuongViRow>;
    IsDefault: Column<SanPhamHuongViRow>;
}

export class SanPhamHuongViColumns extends ColumnsBase<SanPhamHuongViRow> {
    static readonly columnsKey = 'SanPhamDB.SanPhamHuongVi';
    static readonly Fields = fieldsProxy<SanPhamHuongViColumns>();
}